Extensions/xkit_patches.js

```diff
@@ -1118,6 +1120,180 @@ XKit.extensions.xkit_patches = new Object({
+						width: 88%;
+						padding-left: 10px;
+					}
+
+					.xkit--react .controls_section a {
+						text-decoration: none;
+					}
+
+					.xkit--react .controls_section a .count {
+						position: absolute;
+						top: 0;
+						right: 10px;
+						font-size: 13px;
+						font-weight: 400;
+						display: flex;
+					}
+
+					.xkit--react .small_links {
+						border-top: 1px solid var(--transparent-white-7);
+						overflow: hidden;
+					}
+
+					.xkit--react .small_links a {
+						padding: 11px 13px 0;
+						color: var(--transparent-white-65);
+						font-size: 11px;
+					}
+
+					.xkit--react .small_links a:first-child { float: left; }
+					.xkit--react .small_links a:nth-child(2) { float: right; }
+			`,
+
+				init: async function() {
+					await XKit.css_map.getCssMap();
+
+					if ($("#xkit_react_sidebar").length) { return; }
+					//everything after this check is synchronous
+
+					if (!this.css_added) {
+						XKit.tools.add_css(this.react_sidebar_css, "xkit_patches");
+						this.css_added = true;
+					}
+
+					const html = `<div id="xkit_react_sidebar"></div>`;
+
+					//inject the xkit sidebar after the navigation on tumblr.com/blog/myblogname pages
+					const $navSidebarItem = $(XKit.css_map.keyToCss("sideBar")).first().parent();
+					if ($navSidebarItem.length) {
+						$navSidebarItem.after(html);
+						return;
+					}
+
+					//otherwise, inject the xkit sidebar at the top
+					const $sidebarContainer = $(XKit.css_map.keyToCss("sidebar")).find("> aside");
+					$sidebarContainer.prepend(html);
+
+
+					//fix for "after navigation" race condition:
+					//detect if we loaded before the sidebar navigation and wait for it
+					const $loadingPlaceholder = $(XKit.css_map.descendantSelector("sidebar", "loadingPlaceholder"));
+					if ($loadingPlaceholder.length) {
+
+						const loadingObserver = new MutationObserver(function(mutations) {
+							const $newNavSidebarItem = $(XKit.css_map.keyToCss("sideBar")).first().parent();
+							if ($newNavSidebarItem.length) {
+								$newNavSidebarItem.after($("#xkit_react_sidebar"));
+								loadingObserver.disconnect();
+							}
+						});
+						loadingObserver.observe($sidebarContainer.get(0), {childList: true, subtree: true});
+					}
+					this.running = false;
+				},
+
+				/**
+				 * Constructs HTML to add to the sidebar with react-specific CSS.
+				 * Primarily used by add, but can be used directly for custom positioning.
+				 * @param {Object} section
+				 * @param {String} section.id - The element ID for the whole sidebar section
+				 * @param {String} [section.title] - Visible header text of the sidebar section
+				 * @param {Object[]} [section.items] - Array of objects containing button data
+				 * @param {String} section.items[].id - Button element ID
+				 * @param {String} section.items[].text - Visible button text
+				 * @param {Number/String} [section.items[].count] - Text to be displayed as a counter on the button
```

Comment:
Another nitpick that doesn't require changing: Number|String is more commonly used. Technically in a type annotation like this it would be `number|string` since Number and String refer to the boxed versions of the primitive number and string types.

Reply:
Yeah, all of the JSDoc—actually, most of the green code here—was copy-paste :D That did look odd though.